Pass CMAKE_CXX_FLAGS through to the add_custom_command invocation of the compiler in usFunctionEmbedResources.cmake

moraaljj opened the pull request:
Updating usFunctionEmbedResources.cmake to use CMAKE_CXX_FLAGS rather than building the flags itself. This should allow changes to compilation flags in other cmake files to affect the compiled empty resource file for Darwin-based platforms.

karthikreddy09 commented on cmake/usFunctionEmbedResources.cmake:
```diff
@@ -154,21 +154,22 @@ function(usFunctionEmbedResources)
+       #  >> clang ... -c ... -o stub.o
+       # (pardon the elipsis for abbreviation)
+       separate_arguments(TEMP_CXX_FLAGS UNIX_COMMAND ${CMAKE_CXX_FLAGS})
+       foreach(flag ${TEMP_CXX_FLAGS})
```
Any reason why this for loop is needed? passing the output of separate_arguments to add_custom_command works fine for me.

```
add_custom_command(
        OUTPUT ${_source_output}
        COMMAND ${CMAKE_CXX_COMPILER} ${TEMP_CXX_FLAGS} -c ${US_CMAKE_RESOURCE_DEPENDENCIES_CPP} -o stub.o
        COMMAND ${CMAKE_LINKER} -r -sectcreate __TEXT us_resources ${_zip_archive_name} stub.o -o ${_source_output}
        DEPENDS ${_zip_archive}
        WORKING_DIRECTORY ${_zip_archive_path}
        COMMENT "Linking resources zip file for ${US_RESOURCE_TARGET}"
        VERBATIM
       )
```

moraaljj commented:
Hi,

The loop is necessary due to an odd behavior with add_custom_command and
passing in CMAKE_CXX_FLAGS using the -D option to cmake.  The loop ensures
that the quotation marks around the arguments are removed so they can be
properly parsed.

-J

On Tue, Aug 8, 2017, 11:22 Abhinay Reddyreddy <notifications@github.com>
wrote:

> *@karthikreddy09* requested changes on this pull request.
> ------------------------------
>
> In cmake/usFunctionEmbedResources.cmake
> <https://github.com/CppMicroServices/CppMicroServices/pull/226#discussion_r131941892>
> :
>
> > -      endif()
> -      if(CMAKE_OSX_ARCHITECTURES)
> -        list(APPEND US_OSX_CXX_FLAGS -arch ${CMAKE_OSX_ARCHITECTURES})
> -      endif()
> -      if(CMAKE_OSX_DEPLOYMENT_TARGET)
> -        list(APPEND US_OSX_CXX_FLAGS -mmacosx-version-min=${CMAKE_OSX_DEPLOYMENT_TARGET})
> -      endif()
> +        # This resolves an issue where passing in CMAKE_CXX_FLAGS using
> +       # -D CMAKE_CXX_FLAGS:STRING="..." results in a quoted string being passed to
> +       # the command below which results in the command being:
> +       #  >> clang "..." -c ... -o stub.o
> +       # when it should really be:
> +       #  >> clang ... -c ... -o stub.o
> +       # (pardon the elipsis for abbreviation)
> +       separate_arguments(TEMP_CXX_FLAGS UNIX_COMMAND ${CMAKE_CXX_FLAGS})
> +       foreach(flag ${TEMP_CXX_FLAGS})
>
> Any reason why this for loop is needed? passing the output of
> separate_arguments to add_custom_command works fine for me.
>
> add_custom_command(
>         OUTPUT ${_source_output}
>         COMMAND ${CMAKE_CXX_COMPILER} ${TEMP_CXX_FLAGS} -c ${US_CMAKE_RESOURCE_DEPENDENCIES_CPP} -o stub.o
>         COMMAND ${CMAKE_LINKER} -r -sectcreate __TEXT us_resources ${_zip_archive_name} stub.o -o ${_source_output}
>         DEPENDS ${_zip_archive}
>         WORKING_DIRECTORY ${_zip_archive_path}
>         COMMENT "Linking resources zip file for ${US_RESOURCE_TARGET}"
>         VERBATIM
>        )
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/CppMicroServices/CppMicroServices/pull/226#pullrequestreview-54970403>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AF8dmIMdI0WWe6WaPy4K0tTyiB0Wl06Bks5sWH0ogaJpZM4OwrIO>
> .
>

saschazelzer commented on cmake/usFunctionEmbedResources.cmake:
```diff
@@ -154,21 +154,22 @@ function(usFunctionEmbedResources)
+       # when it should really be:
+       #  >> clang ... -c ... -o stub.o
+       # (pardon the elipsis for abbreviation)
+       separate_arguments(TEMP_CXX_FLAGS UNIX_COMMAND ${CMAKE_CXX_FLAGS})
```
Style: please use `_temp_cxx_flags`

saschazelzer commented on cmake/usFunctionEmbedResources.cmake:
```diff
@@ -154,21 +154,22 @@ function(usFunctionEmbedResources)
+       # (pardon the elipsis for abbreviation)
+       separate_arguments(TEMP_CXX_FLAGS UNIX_COMMAND ${CMAKE_CXX_FLAGS})
+       foreach(flag ${TEMP_CXX_FLAGS})
+         set(US_RESOURCE_CXX_FLAGS ${US_RESOURCE_CXX_FLAGS} ${flag})
```
Style: please use `_resource_cxx_flags`

moraaljj commented:
Hi, I updated and removed the for loop as suggested by Abhinay.  It appears to have the same effect and cmake appears to expand the list automatically in the add_custom_command call.

saschazelzer commented:
Thanks!
